### PR TITLE
[Task][68346] Handle Event Search Cell

### PR DESCRIPTION
**Pull request by TuanKieuDev**
### [Redmine link]
https://edu-redmine.sun-asterisk.vn/issues/68346

#### Screenshots (nếu có)

https://github.com/awesome-academy/ios_intern_kieu_anh_tuan-b_project_1/assets/56763942/45635652-1a71-4025-9c2c-7644e77b3590

### CODE REVIEW

File: MarvelApp/MarvelApp/View/Cell/SearchTableViewCell.swift [hunk 1]
```diff
@@ -26,6 +38,10 @@ final class SearchTableViewCell: UITableViewCell {
                 self?.resultImageView.image = image
             }
         }
+    }
 
+    @objc
+    func resultViewTapped(sender: UITapGestureRecognizer) {
```
Comment (hlinh88) [inline MarvelApp/MarvelApp/View/Cell/SearchTableViewCell.swift]:
```suggestion
    private func resultViewTapped(sender: UITapGestureRecognizer) {
```

File: MarvelApp/MarvelApp/View/Cell/SearchTableViewCell.swift [hunk 2]
```diff
@@ -11,9 +11,21 @@ final class SearchTableViewCell: UITableViewCell {
...
+    }
+
+    private func makeViewTappable() {
+        let gesture = UITapGestureRecognizer(target: self, action: #selector(resultViewTapped))
+        resultView.isUserInteractionEnabled = true
+        resultView.addGestureRecognizer(gesture)
+
     }
 
     func configure(_ result: OverviewInformation) {
```
Comment (hlinh88) [inline MarvelApp/MarvelApp/View/Cell/SearchTableViewCell.swift]:
```suggestion
   private func configure(_ result: OverviewInformation) {
```

File: MarvelApp/MarvelApp/Controller/Favorite/FavoriteViewController.swift [hunk 3]
```diff
@@ -90,11 +90,18 @@ extension FavoriteViewController: UITableViewDataSource {
...
+            let overviewInformation = OverviewInformation(
+                id: Int(item.id),
+                thumbnail: Thumbnail(path: item.thumbnailPath ?? Constant.emptyString,
+                                     extension: item.thumbnailExtension ?? Constant.emptyString),
+                name: item.title)
+            cell.configure(overviewInformation)
+            cell.goDetail = { [weak self] in
+                let detailVC = InformationDetailViewController()
+                detailVC.setInformationData(overviewInformation: overviewInformation)
+                self?.navigationController?.pushViewController(detailVC, animated: true)
```
Comment (ghost) [inline MarvelApp/MarvelApp/Controller/Favorite/FavoriteViewController.swift]: unwrap self?.navigationController? trước, sau đó mới push ấy

File: MarvelApp/MarvelApp/Controller/Search/SearchViewController.swift [hunk 4]
```diff
@@ -94,6 +95,11 @@ extension SearchViewController: UITableViewDataSource {
         if let cell = searchTableView.dequeueReusableCell(
             withIdentifier: "search", for: indexPath) as? SearchTableViewCell {
             cell.configure(searchResult[indexPath.row])
+            cell.goDetail = { [weak self] in
+                let detailVC = InformationDetailViewController()
+                detailVC.setInformationData(overviewInformation: self?.searchResult[indexPath.row])
+                self?.navigationController?.pushViewController(detailVC, animated: true)
```
Comment (ghost) [inline MarvelApp/MarvelApp/Controller/Search/SearchViewController.swift]: .

**Pull request merged**